forte/data/multi_pack.py

```diff
@@ -176,6 +176,172 @@ def get_span_text(self, begin: int, end: int):
             "specific data pack to get text."
         )
 
+    def remove_pack(
+        self, index_of_pack: int, clean_invalid_entries: bool = False
+    ) -> bool:
+        """
+        Remove a data pack at index `index_of_pack` from this multi pack.
+
+        In a multi pack, the data pack to be removed may be associated with
+        some multi pack entries, such as `MultiPackLinks` that are connected
+        with other packs. These entries will become dangling and invalid,
+        thus need to be removed. One can consider removing these links before
+        calling this function, or set the `clean_invalid_entries` to `True` so
+        that they will be automatically pruned.
+
+        Args:
+            index_of_pack (int): The index of pack for removal from the
+              multi pack. If invalid, no pack will be deleted.
+            clean_invalid_entries (bool): Switch for automatically cleaning
+              the entries associated with the data pack being deleted which
+              will become invalid after the removal of the pack. Default is
+              False.
+
+        Returns:
+            True if successful
+
+        Exceptions:
+            if clean_invalid_entries is set to False and the DataPack to be
+            removed have entries (in links, groups) associated with it,
+            ValueError will be raised.
+
+        """
+        pack = self.get_pack_at(index_of_pack)
+
+        if pack is None or (not isinstance(pack, DataPack)):
+            type_name = "None"
+            if pack is not None:
+                type_name = type(pack)
+            raise ValueError(
+                f"Object for the index should be pack, but got "
+                f"type: {type_name}"
+            )
+
+        return self._remove_pack(pack, index_of_pack, clean_invalid_entries)
+
+    def _remove_pack(
+        self,
+        pack: DataPack,
+        index_of_pack: int,
+        clean_invalid_entries: bool = False,
+    ) -> bool:
+        """
+        Remove an existing data pack in the multi pack. To prevent index of the
+        packs following it being changed, set this empty position to None
+        in order to keep the index for the packs intact. in _pack_ref[] and
+        _packs[] the position will be set to None while in _pack_names[] the
```

Comment:
May consider wrap the docstring with backtips around variable names like `_pack_names[]`